edx_sga/sga.py

```diff
@@ -862,7 +863,7 @@ def download(self, path, mime_type, filename, require_staff=False):
             return Response(
                 app_iter=file_contents_iter(path),
                 content_type=mime_type,
-                content_disposition="attachment; filename=" + filename.encode('utf-8')
+                content_disposition="attachment; filename=" + urllib.quote(filename.encode('utf-8'))
```

Comment:
In Firefox this appears to produce a filename with the encoded parts still in it. It looks like you need to modify this slightly to get this to work: https://stackoverflow.com/questions/93551/how-to-encode-the-filename-parameter-of-content-disposition-header-in-http#answer-6745788